src/Generator/Passes/ParamTypeToInterfacePass.cs

```diff
@@ -7,20 +7,22 @@ public class ParamTypeToInterfacePass : TranslationUnitPass
     {
         public override bool VisitFunctionDecl(Function function)
         {
-            if (!function.IsOperator)
-            {
+            if (!function.IsOperator || function.Parameters.Count > 1)
                 ChangeToInterfaceType(function.ReturnType);
```

Comment:
It's not related to the current issue but now that you're changing the pass anyway, please use function.OriginalReturnType instead.